### ISSUE TITLE
TreeView should respond to Space keyboard event

### ISSUE THREAD

**Issue by lukeed** [primer/react#4269]
### Description

Because multiple SubTrees can be expanded, it's [recommended](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role#multi-select_keyboard_interactions) to allow <kbd>Space</kbd> to toggle expansion state. 

While there are additional recommended keyboard navigations, these others aren't as common. Using <kbd>Space</kbd> seems to be supported everywhere.

### Steps to reproduce

1. Visit any `TreeView` instance; [see examples](https://primer.style/react/TreeView#examples)
2. Activate keyboard focus into TreeView
3. Focus a directory item & verify that left/right arrows and <kbd>Enter</kbd> are working
4. Press <kbd>Space</kbd> –> no toggle

### Version

35.29.0 & latest

### Browser

Chrome, Safari, Firefox

**Comment by siddharthkp** [comment]
Hi! 

Luckily, the interactions in TreeView are documented here: https://github.com/github/primer/blob/main/apis/tree-view-api.md#keyboard-interaction

> Because multiple SubTrees can be expanded, it's [recommended](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role#multi-select_keyboard_interactions) to allow Space to toggle expansion state.

I'll confirm with our accessibility experts if we are missing the control for `Space` you suggested ❤

**Comment by siddharthkp** [comment]
Tagging @ichelsea as the first-responder for this week ❤ 👋
